files/incremental_upload.py

```diff
@@ -467,9 +468,9 @@ def main():
         if not isinstance(input_dict, dict):
             raise_error("Expected a dict for downstream input. Got %s." %input_dict)
 
-        for k, v in input_dict.items():
-            if not ((isinstance(k, str) or isinstance(k, basestring)) and
-                    (isinstance(v, str) or isinstance(v, basestring) or isinstance(v, dict))):
+        for k, v in list(input_dict.items()):
+            if not ((isinstance(k, str) or isinstance(k, str)) and
```

Comment:
`(isinstance(k, str) or isinstance(k, str))` ... no need to use OR, right?  basestring is no longer available in python3 I believe.

Reply:
yes, no need to use the same condition twice. again, remnant of 2to3, replaced basestring by str.